### PR TITLE
PredictiveRepresentationLearner

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/algorithms/predictive_representation_learner.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+import gin
+import torch
+
+import alf
+from alf.algorithms.algorithm import Algorithm
+from alf.data_structures import AlgStep, TimeStep, Experience, LossInfo, namedtuple
+from alf.experience_replayers.replay_buffer import BatchInfo, ReplayBuffer
+from alf.nest.utils import convert_device
+from alf.networks import Network, LSTMEncodingNetwork
+from alf.utils.losses import element_wise_squared_loss, element_wise_huber_loss
```
Comment (Haichao-Zhang) [inline alf/algorithms/predictive_representation_learner.py]: ```element_wise_squared_loss``` and ```huber_loss``` are not used

File: alf/algorithms/predictive_representation_learner.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+@gin.configurable
+class PredictiveRepresentationLearner(Algorithm):
+    """Learn representation based on the prediction of future values.
+
+    ``PredictiveRepresentationLearner`` contains 3 ``Network``s:
+    * encoding_net: it is used to encode the raw observation to a latent vector.
+    * dynamics_net: it is used to generate the future latent states from the
+        current latent state.
+    * decoder: it is used to decode the target values and calcuate the loss.
```
Comment (Haichao-Zhang) [inline alf/algorithms/predictive_representation_learner.py]: Might be better to make it clear here that decoder is not purely a ``Network`` but an ``Algorithm``.

File: alf/algorithms/predictive_representation_learner.py (hunk 1 above)
Comment (emailweixu) [reply alf/algorithms/predictive_representation_learner.py]: removed

File: alf/algorithms/predictive_representation_learner.py (hunk 2 above)
Comment (emailweixu) [reply alf/algorithms/predictive_representation_learner.py]: updated

**Review by Haichao-Zhang — APPROVED** [review]

**Pull request merged**